src/utils/createReducer.ts

```diff
@@ -0,0 +1,5 @@
+export const createReducer = <S>(
```

Comment:
I think the typing can be improved so we can type the action too and validate that whatever action is passed is valid. But it's fine for the time being